drakrun/drakrun/main.py

```diff
@@ -290,37 +292,53 @@ def generate_graphs(self, workdir):
         os.unlink(os.path.join(workdir, 'drakmon.csv'))
 
     def slice_logs(self, workdir):
-        plugin_fd = {}
-
-        with open(os.path.join(workdir, 'drakmon.log'), 'rb') as f:
-            while True:
-                line = f.readline()
+        workdir = Path(workdir)
+        log_path = workdir / 'drakmon.log'
+        error_path = workdir / 'parse_errors.log'
 
-                if not line:
-                    break
+        plugin_fd = {}
+        failures = Counter()
 
+        with open(log_path, 'rb') as drakvuf_log, \
+                open(error_path, 'wb') as parse_errors:
+            for line in drakvuf_log:
                 try:
-                    line_s = line.strip().decode('utf-8')
+                    line_s = line.strip().decode()
                     obj = json.loads(line_s)
-                except UnicodeDecodeError:
-                    self.log.exception("BUG: Failed to decode UTF-8 from drakmon.log line: {}".format(line))
-                except json.JSONDecodeError:
-                    self.log.exception("BUG: Failed to JSON parse drakmon.log line: {}".format(line))
 
-                if 'Plugin' not in obj:
-                    obj['Plugin'] = 'unknown'
+                    plugin = obj.get('Plugin', 'unknown')
+
+                    if plugin not in plugin_fd:
+                        plugin_fd[plugin] = open(workdir / f'{plugin}.log', 'w')
+                    else:
+                        plugin_fd[plugin].write('\n')
+
+                    plugin_fd[plugin].write(json.dumps(obj))
+                except (UnicodeDecodeError, json.JSONDecodeError):
+                    # Log the failure and count statistics
+
+                    plugin_heuristic: bytes = r'"Plugin": "(\w+)"'.encode()
+                    match = re.match(plugin_heuristic, line)
+                    if match:
+                        # we've matched a unicode word, this shouldn't fail
+                        plugin = match.group(1).decode()
```

Comment:
better to be on the safe side

```suggestion
                        plugin = match.group(1).decode('utf-8', 'replace')
```